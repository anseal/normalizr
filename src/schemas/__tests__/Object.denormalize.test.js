// eslint-env jest
import { denormalize, schema } from '../../'

describe(`${schema.Object.name} denormalization`, () => {
	test('denormalizes an object', () => {
		expect(true).toBe(true)
		// const userSchema = new schema.Entity('user')
		// const object = new schema.Object({
		// 	user: userSchema,
		// })
		// const entities = {
		// 	user: {
		// 		1: { id: 1, name: 'Nacho' },
		// 	},
		// }
		// expect(denormalize({ user: 1 }, object, entities)).toMatchSnapshot()
	})

	// test('denormalizes plain object shorthand', () => {
	// 	const userSchema = new schema.Entity('user')
	// 	const entities = {
	// 		user: {
	// 			1: { id: 1, name: 'Jane' },
	// 		},
	// 	}
	// 	expect(denormalize({ user: 1 }, { user: userSchema, tacos: {} }, entities)).toMatchSnapshot()
	// })

	// test('denormalizes an object that contains a property representing a an object with an id of zero', () => {
	// 	const userSchema = new schema.Entity('user')
	// 	const object = new schema.Object({
	// 		user: userSchema,
	// 	})
	// 	const entities = {
	// 		user: {
	// 			0: { id: 0, name: 'Chancho' },
	// 		},
	// 	}
	// 	expect(denormalize({ user: 0 }, object, entities)).toMatchSnapshot()
	// })
})
