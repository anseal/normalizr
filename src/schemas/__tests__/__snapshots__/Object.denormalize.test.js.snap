// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectSchema denormalization denormalizes an object 1`] = `
Object {
  "user": Object {
    "id": 1,
    "name": "Nacho",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes an object that contains a property representing a an object with an id of zero 1`] = `
Object {
  "user": Object {
    "id": 0,
    "name": "Chancho",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes plain object shorthand 1`] = `
Object {
  "user": Object {
    "id": 1,
    "name": "Jane",
  },
}
`;
